### PR TITLE
istioctl: handle DISABLE in TLS types for precheck

### LINKED ISSUE #50243: Pre-check incorrectly highlighting IST0169
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

Using the new `precheck` command we get:

```
Warning [IST0169] (DestinationRule zendesk-controller/zendesk) The configuration "VERIFY_CERTIFICATE_AT_CLIENT" changed in release 1.20: previously, TLS verification was skipped. Set `insecureSkipVerify` if this behavior is desired. Or, install with `--set compatibilityVersion=1.20` to retain the old default.
```

On our destinationrule we explicitly set:

```
      tls:
        caCertificates: /etc/ssl/certs/ca-certificates.crt
        mode: SIMPLE
        sni: whatever.zendesk.com
```

My understanding was that this implicitly enabled TLS verification, so this warning feels incorrect?

### Version

```Text
1.21.0
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes #50243

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM, i think we need a test

**Comment by howardjohn** [comment]
tests will be a close follow-up in https://github.com/istio/istio/issues/50255

**Comment by hanxiaop** [comment]
> tests will be a close follow-up in #50255

Makes sense

**Review by hanxiaop — APPROVED** [review]

File: istioctl/pkg/precheck/precheck.go [hunk 1]
```diff
@@ -310,11 +310,22 @@ func checkDestinationRuleTLS(cli kube.CLIClient, messages *diag.Messages) error
...
 	}
 	checkSNI := func(tls *networking.ClientTLSSettings) bool {
-		return tls != nil && tls.Sni == "" && tls.Mode != networking.ClientTLSSettings_ISTIO_MUTUAL
+		if tls == nil {
+			return false
+		}
+		if tls.Mode == networking.ClientTLSSettings_DISABLE || tls.Mode == networking.ClientTLSSettings_ISTIO_MUTUAL {
+			return false
+		}
+		return tls.Sni == ""
```
Comment (hzxuzhonghu) [inline istioctl/pkg/precheck/precheck.go:328]:
```suggestion
		return tls.Sni == "" && len(tls.SubjectAltNames) == 0 &&  !tls.GetInsecureSkipVerify().GetValue()
```

Comment (howardjohn) [reply istioctl/pkg/precheck/precheck.go:328]: If SNI is empty: we automatically set SNI
If SNI is empty AND SAN is empty: we automatically set SNI and SAN

So the logic I have is correct I think. The behavior is changed even if SAN is explicitly set.

**Comment by howardjohn** [comment]
ping @hzxuzhonghu canyou take a look?

File: istioctl/pkg/precheck/precheck.go (hunk 1 above)
Comment (hzxuzhonghu) [reply istioctl/pkg/precheck/precheck.go:328]: Right, san check is more strict, so just check sni for warning is ok

**Pull request merged**